tensor/interpolateHypercube.cc

```diff
@@ -69,20 +71,28 @@ namespace civita
                 src.dimension.units!=dst.dimension.units) //TODO handle conversion between different units
               throw runtime_error("mismatch between unnamed dimensions");
             interimHC.xvectors.push_back(dst);
-            rotation[i]=i;
+            tmpRotation[i]=i;
           }
         else // find matching names dimension
           {
             auto dst=find_if(targetHC.begin(), targetHC.end(),
                              [&](const XVector& i){return i.name==src.name;});
             if (dst==targetHC.end())
-              throw runtime_error("no matching dimension found");
-            interimHC.xvectors.push_back(*dst);
-            rotation[dst-targetHC.begin()]=i;
+            {
+              // possible alternative when targetHC has no xvectors or undefined ones. for feature 147
+              interimHC.xvectors.push_back(src);
+              tmpRotation[i]=i;
+            }
+            else
+              {
+                interimHC.xvectors.push_back(*dst);
+                tmpRotation[dst-targetHC.begin()]=i;
```

Comment:
using emplace is cheaper here and lines 74 and 84, as you don't need to search for an existing key first. emplace will not overwrite an existing value, but we don't care about that, as we're going to throw in that circumstance.

Reply:
Done, thanks :)

Reply:
Thanks - looks good, at least to visual inspection.